src/evergreen.ts

```diff
@@ -1,235 +1,262 @@
+  }
 
-    /**
-     * Gets the Evergreen banner message
-     *
-     * @param callback - function to process the response
-     * @returns nothing
-     */
-    public getBanner(callback: request.RequestCallback) {
-        this.getAPIResource(callback, apiV2Resource("admin/banner"));
-    }
+  /**
+   * Updates header cookies when given login credentials
+   *
+   * @param callback - function to process the response
+   * @param username - Evergreen user's username
+   * @param password - Evergreen user's password
+   * @returns nothing
+   */
+  public getToken(callback: request.RequestCallback, username?: string, password?: string) {
+    const params = {
+      username: username,
+      password: password,
+    };
+    this.postUIResource(callback, "login", params);
+  }
 
-    /**
-     * Sets the Evergreen banner message
-     *
-     * @param callback - function to process the response
-     * @param message - text to set in the banner
-     * @param theme - color theme to use
-     * @returns nothing
-     */
-    public setBanner(callback: request.RequestCallback, message: string, theme: string) {
-        const body = {
-            banner: message,
-            theme: theme,
-        };
-        this.postAPIResource(callback, apiV2Resource("admin/settings"), body);
-    }
+  /**
+   * Gets patches for a particular user
+   * @param callback - function to process the response
+   * @param username - Evergreen user's username
+   * @returns nothing
+   */
+  public getPatches(callback: request.RequestCallback, username?: string, page?: number) {
+    const resource = "json/patches/user/" + username;
+    const params = {
+      page: page,
+    };
+    this.getUIResource(callback, resource, params);
+  }
 
-    private formRequest(url: string, requireHeaders: boolean, body?: any): requestOpts {
-      let headers = {};
-      if (requireHeaders) {
-        headers = {
-          "Api-User": this.username,
-          "Api-Key": this.key,
-        };
-      }
-      const opts: requestOpts = {
-        headers: headers,
-        url: url,
+  /**
+   * Gets logs of a particular type for a particular task/
+   * @param callback - function to process the response
+   * @param id - identifier for the task whose logs we want
+   * @param type - type of log to return
+   * @returns nothing
+   */
+  public getLogs(callback: request.RequestCallback, url: string) {
+    const textUrl = url + "&text=true";
+    request.get(textUrl, callback);
+  }
+
+  /**
+   * Gets build for a given build ID
+   * @param id - build ID whose tasks we want
+   * @returns nothing
+   */
+  public getBuild(callback: request.RequestCallback, id: string) {
+    const resource = "builds/" + id;
+    this.getAPIResource(callback, apiV2Resource(resource));
+  }
+
+  /**
+   * Gets tasks for a given build ID
+   * @param id - build ID whose tasks we want
+   * @returns nothing
+   */
+  public getTasks(callback: request.RequestCallback, id: string) {
```

Comment:
call this getTasksForBuild and name the parameter buildID (to avoid ambiguity with a task ID)

Reply:
Fixed